backend/parsing/WorkParser.java

```diff
@@ -1,84 +1,112 @@
 package parsing;
 
-import org.json.JSONObject;
-
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.util.UUID;
+import java.util.LinkedList;
+import java.util.List;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
 public class WorkParser {
 
     // Magic constants identifying the different groups in the regular expression.
-    private static final String LANGUAGE_GROUP = "language", EXERCISE_ID_GROUP = "exerciseId", CONTENTS_GROUP = "contents";
+    private static final String LANGUAGE_GROUP = "language", MARKER_HOST_GROUP = "host",
```

Comment:
These could be moved to a separate "WorkParserConstants" or "WorkParserStrings" file.